### PR TITLE
Port documentation to mdx.

### PR DESCRIPTION
**Pull request by tmcgilchrist**

### CODE REVIEW

File: doc/dune [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+(mdx 
+  (packages current current_git current_docker)
```
Comment (talex5) [inline doc/dune:3]: Probably also needs a `(package current_examples)` here, so it doesn't try to build it for the other packages. Might need to depend on dune 2.9 for that.

File: doc/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+Welcome to the OCurrent wiki!
```
Comment (talex5) [inline doc/index.md:1]: Not a wiki now!

File: doc/index.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+### Introduction
+
+* [Example pipelines](example_pipelines.md)
+* [Skeleton project](https://github.com/ocurrent/ocurrent-skeleton) - a template for new projects
+* [Internals](internals.md) - how it all works
```
Comment (talex5) [inline doc/index.md:7]: Broken link.

File: doc/index.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+* [API docs](https://ocurrent.github.io/ocurrent/index.html)
+
+### OCurrent plugins
+
+* [Current_docker](https://ocurrent.github.io/ocurrent/current_docker/index.html)
+* [Current_fs](https://ocurrent.github.io/ocurrent/current_fs/index.html)
```
Comment (talex5) [inline doc/index.md:13]: Broken link.

File: doc/index.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+* [Current_git](https://ocurrent.github.io/ocurrent/current_git/index.html)
+* [Current_github](https://ocurrent.github.io/ocurrent/current_github/index.html)
+* [Current_slack](https://ocurrent.github.io/ocurrent/current_slack/index.html)
+* [Current_rpc](https://ocurrent.github.io/ocurrent/current_rpc/index.html) - control OCurrent remotely via Cap'n Proto
+
+Or you can [write your own plugin](writing_plugins.md).
+
+### Design and implementation details
+
+* [[The on-disk cache format | Disk cache]]
```
Comment (talex5) [inline doc/index.md:23]: Broken link.

File: examples/build_matrix.ml [hunk 6]
```diff
@@ -18,6 +18,8 @@ let dockerfile ~base ~ocaml_version =
   copy ~src:["."] ~dst:"/src/" () @@
   run "opam install -tv ."
 
+[@@@part "pipeline"]
```
Comment (talex5) [inline examples/build_matrix.ml:21]: Might be worth adding a comment here explaining what this is for.

File: examples/docker_build_local.ml [hunk 7]
```diff
@@ -9,12 +9,15 @@ let timeout = Duration.of_min 50    (* Max build time *)
 
 let () = Prometheus_unix.Logging.init ()
 
+[@@@part "pipeline"]
```
Comment (talex5) [inline examples/docker_build_local.ml:12]: Might be worth adding a comment here explaining what this is for.

**Review by talex5 — COMMENTED** [review]
CI is complaining about duplicate rules. It might be easier to move the examples inside the doc directory.

File: doc/index.md (hunk 4 above)
Comment (tmcgilchrist) [reply doc/index.md:13]: `Current_fs` doesn't get exposed as a library with an opam file.
Should it?

Comment (talex5) [reply doc/index.md:13]: We might split it out later: see https://github.com/ocurrent/ocurrent/pull/276#issuecomment-867484381

Comment (tmcgilchrist) [reply doc/index.md]: I'll point to the code and this can be updated when that happens.

File: doc/index.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+* [Example pipelines](example_pipelines.md)
+* [Skeleton project](https://github.com/ocurrent/ocurrent-skeleton) - a template for new projects
+* [Internals](internals.md) - how it all works
+* [API docs](https://ocurrent.github.io/ocurrent/index.html)
+
+### OCurrent plugins
+
+* [Current_docker](https://ocurrent.github.io/ocurrent/current_docker/index.html)
+* [Current_fs](https://github.com/ocurrent/ocurrent/blob/master/plugins/fs/current_fs.mli)
```
Comment (talex5) [inline doc/index.md:13]:
```suggestion
* [Current_fs](https://ocurrent.github.io/ocurrent/current/Current_fs/index.html)
```

**Review by talex5 — COMMENTED** [review]
Thanks! Some nit-picking inline :-)

File: doc/dune [hunk 9]
```diff
@@ -0,0 +1,6 @@
+
+(mdx 
+ (packages current_examples)
+ (preludes prelude.ml)
```
Comment (talex5) [inline doc/dune:4]: I suggest getting rid of this predule. It contains things the user actually needs to type, so hiding it isn't helpful.

File: doc/internals.md [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+- The `Current_incr.map (Dyn.map ~id f) x.v` is the dynamic part, saying that the value is the result
+  of applying `f` to the dynamic value part of `x` (`x.v`).
+  `Dyn.map` handles the `result` type (only affecting `Ok` values)
+  and the `Current_incr.map` handles the incremental part, so it will update automatically.
+
+`map` operations aren't very interesting (they only appear on diagrams if they fail), but primitive operations
+and some other combinators are shown. We can render the `result` pipeline above like this:
+
+```
+utop # Format.printf "%a@." Term.Analysis.pp result;;
```
Comment (talex5) [inline doc/internals.md:189]:
```suggestion
```ocaml
# Format.printf "@[<h>%a@]@." Term.Analysis.pp result;;
```

File: doc/prelude.ml [hunk 11]
```diff
@@ -0,0 +1 @@
+#require "current,current_git,current_docker,current_github,current.term";;
```
Comment (talex5) [inline doc/prelude.ml:1]: Can just delete this file.

**Comment by talex5** [comment]
Thanks!

**Pull request merged**